pkg/operator/management/management_state.go

```diff
@@ -29,6 +29,8 @@ func SetOperatorAlwaysManaged() {
 // SetOperatorNotRemovable is one time choice the operator author can make to indicate the operator does not support
 // removing of his operand. This makes sense for operators like kube-apiserver where removing operand will lead to a
 // bricked, non-automatically recoverable state.
+// Unless explicitly set, an operator supports Removed state.
+// Note that most library-go controllers do not support Removed state, please check each controller separately.
 func SetOperatorNotRemovable() {
 	allowOperatorRemovedState = false
```

Comment:
wow, we made a global?  really?  Please fix while you're in here.

Reply:
That's a bigger refactoring. Will try something (`OperatorType` struct with `IsOperatorAlwaysManaged()` and similar functions?)

Reply:
I am afraid this may be too big for this PR. `IsOperatorManaged` is called deep in other controllers, for example in this "base" controller that's used in 5 other encryption controllers

https://github.com/openshift/library-go/blob/17708d019f9cb170ff1f31265d5acfa82b9a82e9/pkg/operator/encryption/controllers/controller.go#L30-L30

Passing `OperatorType` down there seems possible, but it will be quite bug PR on its own.